great_expectations/self_check/util.py

```diff
@@ -467,6 +467,24 @@
 # except ImportError:
 #     teradatasqlalchemy = None
 
+try:
+    from great_expectations.dataset import SqlAlchemyDataset
+    from great_expectations.dataset.sqlalchemy_dataset import SqlAlchemyBatchReference
+except ImportError:
+    SqlAlchemyDataset = None  # type: ignore[misc,assignment] # could be None
+    SqlAlchemyBatchReference = None  # type: ignore[misc,assignment] # could be None
+    logger.debug(
+        "Unable to load sqlalchemy dataset; install optional sqlalchemy dependency for support."
+    )
+
+try:
+    from great_expectations.dataset import SparkDFDataset
+except ImportError:
+    SparkDFDataset = None  # type: ignore[misc,assignment] # could be None
```

Comment:
Why are we changing this to a try:except: import?

If we are doing type-checking on these they need to be available during the type-checking step.

Reply:
@Kilo59 I believe that this is important for when the corresponding `Spark` and `SQLAlchemy` imports are not present.